### ISSUE TITLE
twisted.names resolver does not cope with results returned in a different case than what was queried

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8343]
|[<img alt="zaheerm's avatar" src="https://avatars.githubusercontent.com/u/68943?s=50" width="50" height="50">](https://github.com/zaheerm)| @zaheerm reported|
|-|-|
|Trac ID|trac#8343|
|Type|defect|
|Created|2016-05-21 01:29:20Z|
|Branch|https://github.com/twisted/twisted/tree/8343-zaheerm-dns_case_insensitive|

Example, if you search for boo.hoo.com and the DNS server returns an A record for boo.hoo.COM, the resolver will fail.

DNS is meant to be case insensitive.

Pull request is here: https://github.com/twisted/twisted/pull/63

<details><summary>Searchable metadata</summary>

```
trac-id__8343 8343
type__defect defect
reporter__zaheerm zaheerm
priority__normal normal
milestone__None None
branch__8343_zaheerm_dns_case_insensitive 8343-zaheerm-dns_case_insensitive
branch_author__ 
status__closed closed
resolution__fixed fixed
component__names names
keywords__None None
time__1463794160024628 1463794160024628
changetime__1464182213493278 1464182213493278
version__None None
owner__zaheerm zaheerm

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban set owner to @zaheerm|
|-|-|

Many thanks for you contribution.

1. All new changes require updated test to make sure we don't end up with regression and the test should document the intended behaviour.

2. Also, since this is an ugly bug, it would be nice if this will be advertised in the release notes of the next release. To prevent merge conflict release notes are broken into fragments. For more details see http://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles

--------

Please update your branch and resubmit for review.

Many thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="zaheerm's avatar" src="https://avatars.githubusercontent.com/u/68943?s=50" width="50" height="50">](https://github.com/zaheerm)<a name="note_2"></a>|@zaheerm removed owner|
|-|-|

Branch updated to include unit test and topfile.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban commented|
|-|-|

The changes looks good.

I think that the docstring if the test can be improved. see this for more info https://jml.io/pages/test-docstrings.html

Basically

```
Test comparisons between L{Name} instances, including validating
case insensitivity
```

does not tell me much about the test scenario and what is the expected behaviour.

For me, a better docstring would be.

```
L{Name} instances are equal as long as they have the same value for L{Name.name},
regardless of the case.
```

Please check my comments...

meanwhile I will try to find out what is the process for pulling your changes

**Comment by twisted-trac** [comment]
|[<img alt="zaheerm's avatar" src="https://avatars.githubusercontent.com/u/68943?s=50" width="50" height="50">](https://github.com/zaheerm)<a name="note_4"></a>|@zaheerm commented|
|-|-|

I updated the docstring in the branch as per your review comment.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @zaheerm|
|-|-|


What are the print calls from latest commit ? https://github.com/twisted/twisted/pull/63/commits/16ed0c899702855ab8062e40744110675187873f

---------

and the change to lookupAddress ?

Looks like unrelated changes.

If the change to lookupAddress is related to this, please write a test to cover what was broken before :)

And the test docstrings are not updated :)

Don't worry... it happens and this is why we have the code review process

Please update and put for review.

**Comment by twisted-trac** [comment]
|[<img alt="zaheerm's avatar" src="https://avatars.githubusercontent.com/u/68943?s=50" width="50" height="50">](https://github.com/zaheerm)<a name="note_6"></a>|@zaheerm removed owner|
|-|-|

Updated...no it was me messing around with something else, apologies! It shouldn't have ever made it into git!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_7"></a>|@adiroiban set owner to @zaheerm|
|-|-|

Thanks for the update.

The patch looks good.

I am waiting for the tests to pass and will merge.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset e08fb4566f7cc0a118fc39b55842ceb4d11b0ab2
```
#!CommitTicketReference repository="" revision="e08fb4566f7cc0a118fc39b55842ceb4d11b0ab2"
Merge 8343-zaheerm-dns_case_insensitive: twisted.names.dns.Name comparison is now case insensitive.

Author: zaheerm
Reviewer: adiroiban
Fixes: #8343
```

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_9"></a>|@adiroiban commented|
|-|-|

I have merged this based on the initial submission.

if t.n.dns.Name.name attribute should be case insensitive, I would argue that it should be turned into a property which will always convert that value in lower case.

Also, beside the current unit tests, it would be great to have a high level function tests which demonstrate how the twisted.names.client.Resolver behavior if fixed by this branch... and how it was failing in previous trunk.

zaheerm if you would like to contribute to Twisted and make sure that twisted.names.client.Resolver will always work in the future, please consider submitting a patch with the updated tests.

Thanks for your contribution!
